modules/javafx.graphics/src/main/java/com/sun/javafx/scene/layout/ScaledMath.java

```diff
@@ -0,0 +1,79 @@
+package com.sun.javafx.scene.layout;
+
+/**
+ * Math functions which deal with rounding scaled values.
+ */
+public class ScaledMath {
+
+    /**
+     * The value is floored for a given scale using Math.floor.
+     * When the absolute value of the given value multiplied by the
+     * current scale is less than 10^15, then this method guarantees that:
+     *
+     * <pre>scaledFloor(scaledFloor(value, scale), scale) == scaledFloor(value, scale)</pre>
+     *
+     * The limit is about 10^15 because double values will no longer be able to represent
+     * larger integers with exact precision beyond this limit.
+     *
+     * @param value The value that needs to be floored
```

Comment:
is this method only for positive values?

Reply:
No, it can be used for any value.